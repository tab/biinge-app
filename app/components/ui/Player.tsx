import React, { useState } from "react"
import {
  ActivityIndicator,
  SafeAreaView,
  View,
  Pressable,
  StyleSheet,
  Dimensions,
} from "react-native"
import YouTube from "react-native-youtube"

import { YOUTUBE_API_KEY } from "config"
import Icon from "components/ui/Icon"
import colors from "styles/colors"
import { TMDB_VIDEO_NOT_EMBEDDABLE } from "types"

type PlayerProps = {
  videoId: string
  onClose: () => void
}

const PlayerComponent: React.FC<PlayerProps> = ({ videoId, onClose }) => {
  const [loading, setLoading] = useState(true)

  const handleReady = () => {
    setTimeout(() => {
      setLoading(false)
    }, 1500)
  }

  const handleError = (error: { error: string; target: number }) => {
    if (error && error.error === TMDB_VIDEO_NOT_EMBEDDABLE) {
      setLoading(false)
    }
  }

  return (
    <SafeAreaView style={styles.root}>
      <View style={styles.content}>
        <Icon style={styles.icon} name="close" color={colors.white} size={30} />
        <Pressable style={styles.close} onPress={onClose} />
        <YouTube
          apiKey={YOUTUBE_API_KEY}
          videoId={videoId}
          play
          controls={1}
          modestbranding
          showFullscreenButton
          fullscreen
          style={styles.player}
          onReady={handleReady}
          onError={handleError}
        />
        <ActivityIndicator
          style={loading ? styles.loader : styles.hidden}
          animating={true}
          size="small"
          color={colors.white}
        />
      </View>
    </SafeAreaView>
  )
}

const { width } = Dimensions.get("window")

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: colors.darkBlack,
  },
  content: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  player: {
    alignItems: "center",
    justifyContent: "center",
    aspectRatio: "1",
    width: width,
  },
  loader: {
    backgroundColor: colors.darkBlack,
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  hidden: {
    opacity: 0,
  },
  close: {
    backgroundColor: "transparent",
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  icon: {
    position: "absolute",
    top: 5,
    left: 10,
  },
})

export default PlayerComponent
