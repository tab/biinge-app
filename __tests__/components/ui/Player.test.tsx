import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import Player from "components/ui/Player"

describe("Player", () => {
  const videoId = "123456789"

  const handlePlayer = jest.fn()

  it("renders correctly", () => {
    const tree = render(<Player videoId={videoId} onClose={handlePlayer} />)
    expect(tree).toMatchSnapshot()
  })
})
