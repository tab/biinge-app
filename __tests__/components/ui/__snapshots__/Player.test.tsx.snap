// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 30,
          },
          {
            "left": 10,
            "position": "absolute",
            "top": 5,
          },
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î­‹
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <RCTYouTube
      onChangeFullscreen={[Function]}
      onChangeQuality={[Function]}
      onChangeState={[Function]}
      onError={[Function]}
      onProgress={[Function]}
      onReady={[Function]}
      play={true}
      playerParams={
        {
          "playerVars": {
            "controls": 1,
            "fs": undefined,
            "loop": undefined,
            "modestbranding": 1,
            "origin": undefined,
            "playlist": undefined,
            "playsinline": undefined,
            "rel": undefined,
            "showinfo": undefined,
          },
          "playlistId": undefined,
          "videoId": "123456789",
        }
      }
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "alignItems": "center",
            "aspectRatio": "1",
            "justifyContent": "center",
            "width": 750,
          },
        ]
      }
      videoId="123456789"
    />
    <ActivityIndicator
      animating={true}
      color="#ffffff"
      size="small"
      style={
        {
          "backgroundColor": "#000",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
